### PR TITLE
Ensure `controllerManager`'s extra arg `allocate-node-cidrs` is set to true

### PR DESCRIPTION
**Pull request by whites11**
in `KubeadmControlPlane` for Azure clusters.

### Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: policies/azure/AzureCAPI.yaml [hunk 1]
```diff
@@ -34,3 +34,20 @@ spec:
...
+          matchLabels:
+            cluster.x-k8s.io/watch-filter: capi
+    mutate:
+      patchStrategicMerge:
+        spec:
+          kubeadmConfigSpec:
+            clusterConfiguration:
+              controllerManager:
+                extraArgs:
+                  allocate-node-cidrs: "true"
```
Comment (stone-z) [inline policies/azure/AzureCAPI.yaml:53]: Just to double check, is the intent to only default or also override an existing value?

**Review by stone-z — APPROVED** [review]
Seems reasonable to me, just one comment

**Review by MarcelMue — APPROVED** [review]

**Comment by MarcelMue** [comment]
Thank you for adding the test case as well :heart:

File: policies/azure/AzureCAPI.yaml (hunk 1 above)
Comment (whites11) [reply policies/azure/AzureCAPI.yaml:53]: I want to override. The only value that works is "true", anything else will generate a broken cluster

Comment (stone-z) [reply policies/azure/AzureCAPI.yaml:53]: Cool, then this looks perfect

**Pull request merged**